src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.String.cs

```diff
@@ -142,6 +213,38 @@ public static void TryReadTextString_BufferTooSmall_ShouldReturnFalse(string act
             Assert.All(buffer, (b => Assert.Equal(0, '\0')));
         }
 
+        [Theory]
+        [InlineData("ab", "5f41ab40ff")]
+        [InlineData("abbc", "5f41ab41bc40ff")]
+        public static void TryReadByteString_IndefiniteLengthConcatenated_BufferTooSmall_ShouldReturnFalse(string expectedHexValue, string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborReader(data);
+
+            byte[] buffer = new byte[expectedHexValue.Length / 2 - 1];
+            bool result = reader.TryReadByteString(buffer, out int bytesWritten);
+
+            Assert.False(result);
+            Assert.Equal(0, bytesWritten);
+            Assert.All(buffer, (b => Assert.Equal(0, b)));
+        }
+
+        [Theory]
+        [InlineData("ab", "7f62616260ff")]
+        [InlineData("abbc", "7f62616262626360ff")]
+        public static void TryReadTextString_IndefiniteLengthConcatenated_BufferTooSmall_ShouldReturnFalse(string expectedValue, string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborReader(data);
+
+            char[] buffer = new char[expectedValue.Length - 1];
+            bool result = reader.TryReadTextString(buffer, out int charsWritten);
+
+            Assert.False(result);
+            Assert.Equal(0, charsWritten);
+            Assert.All(buffer, (b => Assert.Equal(0, '\0')));
+        }
```

Comment:
I'd recommend making `buffer` be big enough (or oversized), and passing `buffer.AsSpan(1)`.  Then, after the return false, see that passing the exact or oversized buffer succeeds. (Which largely shows that the state isn't modified during `return false`).

Reply:
Done